culturefeed_mailing/culturefeed_mailing.pages.inc

```diff
@@ -347,25 +364,55 @@ function culturefeed_mailing_subscription_update_form() {
  */
 function culturefeed_mailing_subscription_update_form_submit($form, &$form_state) {
 
-  $account = DrupalCultureFeed::getLoggedInUser();
+  if (!empty($form_state['values']['zip'])) {
+    $zip = $form_state['values']['zip'];
+    _culturefeed_mailing_set_postal_to_cookie($form_state['values']['zip']);
+  }
+  else {
+    $zip = '';
+  }
+
+  if (CULTUREFEED_API_LIGHT_ID_ALLOWED) {
```

Comment:
Als een user ingelogd is, zal hij bij submit toch steeds in deze case terecht komen en dus niet meer via  `DrupalCultureFeed::updateUser($user_update, $fields)` ge-update worden maar altijd via `culturefeed_mailing_subscribe_user_light() `.

Moet hier niet een extra controle bij op $account?

Reply:
Er wordt enkel vanuit nieuwsbrieven naar deze pagina verwezen. De kans dat een user aangemeld is als hij op deze pagina terecht komt, is dus zeer klein. En we willen net vermijden dat ze eerst moeten aanmelden om een update van hun postcode te doen.

We willen dus vooral vermijden dat ze op de update pagina komen en naast postcode ook voornaam, achternaam en gemeente moeten aanvullen. Wat mij betreft, mag het zo blijven dus.

Reply:
De edge case valt evenwel wel te vermijden door `if (CULTUREFEED_API_LIGHT_ID_ALLOWED) {` te vervangen door `if (!$account && CULTUREFEED_API_LIGHT_ID_ALLOWED) {` en natuurlijk de account eerst op te halen. 

Hiermee komt een ingelogde gebruiker nog steeds in de case terecht waarbij hij alle velden pre-filled ziet. Zoniet kan de ingelogde gebruiker op die pagina zijn zip/email niet aanpassen omdat hij de de "CODE_MAILING_ALREADY_SUBSCRIBED" error message zal te zien krijgen.

Als je dit niet nodig acht, voer ik de merge zo uit.

Reply:
ik heb het net getest als aangemelde reeds ingeschreven gebruiker en die CODE_MAILING_ALREADY_SUBSCRIBED lijkt niet getriggered te worden (of alleszins niet te verhinderen dat de zipcode geüpdated wordt). Ik zou zo mergen dus